backend/packages/Upgrade/src/api/services/MoocletExperimentService.ts

```diff
@@ -164,13 +165,45 @@ export class MoocletExperimentService extends ExperimentService {
   }
 
   public async syncCreate(params: SyncCreateParams): Promise<ExperimentDTO> {
-    return this.dataSource.transaction((manager) => this.handleCreateMoocletTransaction(manager, params));
+    return this.dataSource.transaction(async (manager) => {
+      const experimentResponse = await this.createUpgradeExperiment(manager, params);
+      params.experimentDTO = experimentResponse;
+      return this.handleCreateMoocletTransaction(manager, params);
+    });
   }
 
-  public async syncUpdate(params: SyncCreateParams): Promise<ExperimentDTO> {
+  public async syncUpdate(params: SyncEditParams): Promise<ExperimentDTO> {
     return this.dataSource.transaction((manager) => this.handleEditMoocletTransaction(manager, params));
   }
 
+  public async syncUpdateWithMoocletAlgorithmTransition(params: SyncEditParams): Promise<ExperimentDTO> {
+    return this.dataSource.transaction(async (manager) => {
+      const updatedExperiment = await this.updateUpgradeExperiment(manager, params);
+
+      // when transitioning away from Mooclet, delete the old refs after successful experiment update
+      if (params.moocletRefToDelete) {
+        await this.moocletExperimentRefRepository.delete(params.moocletRefToDelete.id);
+      }
```

Comment:
Problem: The deletion uses `moocletExperimentRefRepository.delete()` directly instead of using the transaction manager. This means the deletion may not be part of the transaction and could cause:

- Orphaned refs if the transaction rolls back after deletion
- Loss of atomicity

Fix:
```
if (params.moocletRefToDelete) {
  await manager.delete(MoocletExperimentRef, params.moocletRefToDelete.id);
}
```